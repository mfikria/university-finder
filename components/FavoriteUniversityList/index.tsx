import type { NextComponentType } from 'next'
import React, { useEffect } from 'react'
import { List, Typography } from 'antd'
const { Title } = Typography

import { BsFillGeoAltFill, BsBoxArrowUpRight } from 'react-icons/bs'
import { useUniversities } from '../../hooks/university'
import styles from './style.module.scss'
import { University } from '../../types/university.d'
import { useSession } from 'next-auth/react'

const UniversityList: NextComponentType = () => {
  const { data: session } = useSession()

  const { loadingFavorite, favoriteUniversities, fetchFavoriteUniversities } =
    useUniversities()

  useEffect(() => {
    if (session) {
      fetchFavoriteUniversities()
    }
  }, [session])

  return (
    <>
      <Title level={3}>Your Favorite Universities</Title>
      <div id="scrollableList" className={styles.scrollableList}>
        <List
          loading={loadingFavorite}
          dataSource={favoriteUniversities}
          renderItem={(university: University) => {
            const website = university.web_pages[0]
            const description = (
              <>
                <div className={styles.universityDescription}>
                  <BsFillGeoAltFill />
                  {university.country}
                </div>
                <div className={styles.universityWebsite}>
                  <BsBoxArrowUpRight />
                  <a href={website} target="_blank" rel="noreferrer">
                    {website}
                  </a>
                </div>
              </>
            )
            return (
              <List.Item>
                <List.Item.Meta
                  title={university.name}
                  description={description}
                />
              </List.Item>
            )
          }}
        />
      </div>
    </>
  )
}

export default UniversityList
