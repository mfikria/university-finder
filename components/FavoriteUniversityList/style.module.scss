@import "styles/mixins";

.scrollableList {
  height: calc(100vh - 200px);
  padding: 0 16px;
  overflow: auto;
  border: 1px solid rgba(140, 140, 140, 0.35);
  border-top: none;
  background: white;

  :global .ant-list-item {
    position: relative;
  }
}

.universityDescription, .universityWebsite {
  display: inline-flex;
  align-items: center;
  font-size: 12px;
  svg {
    margin-right: 4px;
  }
}

.universityDescription {
  margin-right: 12px;
}
