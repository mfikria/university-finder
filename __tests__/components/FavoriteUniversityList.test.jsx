/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import FavoriteUniversityList from '../../components/FavoriteUniversityList'

const mockSession = {
  user: {
    image: null,
    name: "John",
    email: "john@email.com",
  },
  expires: 123213139,
}

jest.mock('next-auth/react', () => ({
  useSession: jest.fn(() => ({
    data: mockSession,
    status: 'loading'
  }))
}))

describe('FavoriteUniversityList', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<FavoriteUniversityList />)
    expect(asFragment()).toMatchSnapshot()
  })
})
