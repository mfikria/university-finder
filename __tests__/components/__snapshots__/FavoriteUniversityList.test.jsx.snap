// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteUniversityList should match snapshot 1`] = `
<DocumentFragment>
  <h3
    class="ant-typography"
  >
    Your Favorite Universities
  </h3>
  <div
    class="scrollableList"
    id="scrollableList"
  >
    <div
      class="ant-list ant-list-split ant-list-loading"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div>
          <div
            class="ant-spin ant-spin-spinning"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-spin-container ant-spin-blur"
        >
          <div
            style="min-height: 53px;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
